src/transports/http.c

```diff
@@ -1348,7 +1359,10 @@ static int http_receivepack(
 
 	s = (http_stream *)*stream;
 
-	/* Use Transfer-Encoding: chunked for this request */
+	/*
+		Use Transfer-Encoding: chunked for this request
+		Warning: This may make it so the whole body is not available for computing sha256.
```

Comment:
I would assume that we use chunked encoding given this code. Why not buffer the chunked requests so that you can compute the full hash, then send them out afterwards?

Reply:
Will do that as a follow up. Fetch uses non-chunked - cloning and syncing etc.